lib/alephant/broker/response/base.rb

```diff
@@ -48,10 +55,24 @@ def add_no_cache_headers
           log
         end
 
-        def component_not_modified(headers, request_env)
-          return false if headers["Last-Modified"].nil? && headers["ETag"].nil?
+        def add_etag_allow_header
+          headers.merge!({
+            "Access-Control-Expose-Headers" => "ETag"
+          })
+        end
+
+        def self.component_not_modified(headers, request_env)
+          return false if request_env.if_modified_since.nil? && request_env.if_none_match.nil?
+
+          last_modified_match = !request_env.if_modified_since.nil? && headers["Last-Modified"] == request_env.if_modified_since
+          etag_match          = !request_env.if_none_match.nil? &&
+            Alephant::Broker::Response::Base.unquote_etag(headers["ETag"]) == Alephant::Broker::Response::Base.unquote_etag(request_env.if_none_match)
```

Comment:
class methods can call other class methods without the prefix - so just `unquote_etag` will do.